presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -125,6 +127,10 @@
     private final boolean skipTargetCleanupOnRollback;
     private final boolean undoMetastoreOperationsEnabled;
 
+    // Cache lastDataCommitTime for read queries.
+    @GuardedBy("this")
+    private Map<SchemaTableName, List<DateTime>> lastDataCommitTimesForRead = new HashMap<>();
```

Comment:
comment not addressed here: https://github.com/prestodb/presto/pull/17830/files#r898223413 and https://github.com/prestodb/presto/pull/17830/files#r898713406